src/establish-metamask-repository.ts

```diff
@@ -0,0 +1,262 @@
+import { RepositoryFilesystem } from './repository-filesystem';
+import {
+  ensureBranchUpToDateWithRemote,
+  getBranchInfo,
+  getCurrentBranchName,
+} from './repository-utils';
+import type { BranchInfo } from './repository-utils';
+import { resolveRepositoryReference } from './resolve-repository-reference';
+
+const log = createModuleLogger(projectLogger, 'establish-metamask-repository');
+
+type ConfirmedRepository = BranchInfo & {
+  shortname: string;
+  directoryPath: string;
+};
+
+/**
+ * A repository within the MetaMask organization on GitHub which has been cloned
+ * to the local filesystem. More concretely, this could either be a template or
+ * a project that requires linting.
+ */
+export type MetaMaskRepository = ConfirmedRepository & {
+  fs: RepositoryFilesystem;
+};
+
+/**
+ * Information about a repository we know exists on the filesystem.
+ */
+export type ExistingRepository = ConfirmedRepository & {
+  createdAutomatically: boolean;
+};
+
+/**
+ * Ensures that there is a proper repository to lint. A repository may be one of
+ * two things: either A) the "short name" of a known repository under the GitHub
+ * MetaMask organization, or B) the path to a directory on the local filesystem.
+ * In the case of a MetaMask repository, this function takes care of
+ * automatically cloning the repository to a temporary location (unless it has
+ * already been cloned), then bringing the repository up to date with its
+ * default branch.
+ *
+ * @param args - The arguments to this function.
+ * @param args.repositoryReference - Either the name of a MetaMask repository,
+ * such as "utils", or the path to a local Git repository.
+ * @param args.workingDirectoryPath - The directory where this tool was run.
+ * @param args.cachedRepositoriesDirectoryPath - The directory where MetaMask
+ * repositories will be (or have been) cloned.
+ * @param args.outputLogger - Writable streams for output messages.
+ * @returns The repository.
+ */
+export async function establishMetaMaskRepository({
+  repositoryReference,
+  workingDirectoryPath,
+  cachedRepositoriesDirectoryPath,
+  outputLogger,
+}: {
+  repositoryReference: string;
+  workingDirectoryPath: string;
+  cachedRepositoriesDirectoryPath: string;
+  outputLogger: AbstractOutputLogger;
+}): Promise<MetaMaskRepository> {
+  const existingRepository = await ensureRepositoryExists({
+    repositoryReference,
+    workingDirectoryPath,
+    cachedRepositoriesDirectoryPath,
+    outputLogger,
+  });
+  log('Repository is', existingRepository.shortname);
+
+  const repositoryFilesystem = new RepositoryFilesystem(
+    existingRepository.directoryPath,
+  );
+
+  if (existingRepository.createdAutomatically) {
+    await requireDefaultBranchSelected(existingRepository);
+
+    const updatedLastFetchedDate = await ensureBranchUpToDateWithRemote(
+      existingRepository.directoryPath,
+      {
+        remoteBranchName: existingRepository.currentBranchName,
```

Comment:
Was this meant to be this?
```suggestion
        remoteBranchName: existingRepository.defaultBranchName,
```

I'm wondering because prior to the recent commit, it would always sync with the main branch (which is called `defaultBranchName` here). But now it's updating the current branch regardless whether it matches the main branch or not.

Reply:
Yes, good point. This is updated in [ced47f7](https://github.com/MetaMask/module-lint/pull/30/commits/ced47f70de8e140e00e6fe85db17af4d487e0bfc).